python/cuml/cluster/kmeans.pyx

```diff
@@ -453,17 +454,25 @@ class KMeans(Base):
             Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
             ndarray, cuda array interface compliant array like CuPy
 
+        convert_dtype : bool (default = False)
+            When set to True, the predict method will automatically convert
+            the input to the data type which was used to train the model. This
+            will increase memory used for the method.
+
         """
 
         cdef uintptr_t input_ptr
-        X_m, input_ptr, self.n_rows, self.n_cols, self.dtype = \
-            input_to_dev_array(X, order='C')
+        X_m, input_ptr, n_rows, n_cols, self.dtype = \
+            input_to_dev_array(X, order='C', check_dtype=self.dtype,
```

Comment:
So, check_dtype is ignored if convert_to_dtype is active?

Reply:
yes, this was more of a concession given that `input_to_dev_array` is a method outside the class so has no other way of knowing what is the dtype of the class calling it